src/pages/home/report/shouldDisplayNewMarkerOnReportAction.ts

```diff
@@ -0,0 +1,87 @@
+import {isReportActionUnread, isReportPreviewAction, shouldHideNewMarker} from '@libs/ReportActionsUtils';
+import CONST from '@src/CONST';
+import type * as OnyxTypes from '@src/types/onyx';
+
+type ShouldDisplayNewMarkerOnReportActionParams = {
+    /** The reportAction for which the check is done */
+    message: OnyxTypes.ReportAction;
+
+    /** Whether the next message in a list is unread */
+    isNextMessageUnread: boolean;
+
+    /** Is it the earliestReceivedOfflineMessage */
+    isEarliestReceivedOfflineMessage: boolean;
+
+    /** Time for unreadMarker */
+    unreadMarkerTime: string | undefined;
+
+    /** User accountID */
+    accountID: number | undefined;
+
+    /** Map of reportActions saved via usePrev */
+    prevSortedVisibleReportActionsObjects: Record<string, OnyxTypes.ReportAction>;
+
+    /** Current value for vertical offset */
+    scrollingVerticalOffset: number;
+
+    /** Current value for vertical offset */
+    prevUnreadMarkerReportActionID: string | null;
+};
+
+/**
+ * This function decides whether the given report action (message) should have the new marker indicator displayed
+ * It's used for the standard "chat" Report and for `MoneyRequestReport` actions lists.
+ */
+const shouldDisplayNewMarkerOnReportAction = ({
```

Comment:
this function was extracted from `ReportActionsList` almost without change - I'm just passing down arguments so that it can be reused in 2 places.
No logic changes are intended for `ReportActionsList`